src/proj.rs

```diff
@@ -498,6 +500,56 @@ pub struct Proj {
 }
 
 impl Proj {
+    /// Create a coordinate metadata object to be used in coordinate operations.
+    ///
+    /// This creates a coordinate metadata object that can be used in coordinate operations,
+    /// such as transformations. The coordinate metadata object contains information
+    /// about the coordinates, such as the epoch they are referenced to.
+    ///
+    /// # Note
+    /// Only **transformation objects** (e.g. those created by calling [`new()`](fn@Proj::new())) can be
+    /// converted to metadata objects. They cannot be created from pipelines.
+    ///
+    /// # Arguments
+    ///
+    /// * `epoch` - The epoch that the coordinates are referenced to.
+    ///
+    /// # Returns
+    ///
+    /// A new coordinate metadata object or an error if creation failed.
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn coordinate_metadata_create(&self, epoch: f64) -> Result<Proj, ProjCreateError> {
+        // Clone the context to avoid double-free in Drop implementations
+        let cloned_ctx = unsafe { proj_context_clone(self.ctx) };
+
+        let ptr = result_from_create(cloned_ctx, unsafe {
+            proj_coordinate_metadata_create(cloned_ctx, self.c_proj, epoch)
+        })
+        .map_err(|e| ProjCreateError::ProjError(e.message(cloned_ctx)))?;
+
+        Ok(Proj {
+            c_proj: ptr,
+            ctx: cloned_ctx,
+            area: None,
+        })
+    }
+
+    /// Get the epoch from a coordinate metadata object.
+    ///
+    /// This retrieves the epoch associated with a coordinate metadata object.
+    ///
+    /// # Returns
+    ///
+    /// The epoch value as a float, or [`f64::NAN`] if the object doesn't have an associated epoch.
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn coordinate_metadata_get_epoch(&self) -> f64 {
+        unsafe { proj_coordinate_metadata_get_epoch(self.ctx, self.c_proj) }
```

Comment:
Have you had any more luck than me (zero) finding documentation for these methods?

https://github.com/search?q=repo%3AOSGeo%2FPROJ%20proj_coordinate_metadata_get_epoch&type=code

Is `epoch` in any particular units or are those units specific to each projection?

Reply:
"Just" floats of the form year.month,and month can be 0: 2010.3 and 2014.0 are both valid values.

Reply:
I couldn't find it documented in proj, but in gdal, which I'm comfortable assuming has the same semantics:

https://gdal.org/en/stable/api/ogrspatialref.html#_CPPv4NK19OGRSpatialReference18GetCoordinateEpochEv

> coordinateEpoch Coordinate epoch as decimal year (e.g. 2021.3), or 0 if not set, or relevant.

So 2021.3 is April 2021.